### ISSUE TITLE
Using an async selector in another selector, throws an Uncaught promise

### ISSUE THREAD

**Issue by kislakiruben** [facebookexperimental/Recoil#694]
Hi!

I have an async selector which relies on the return value from another async selector. The values are returned fine, but I get an error in the console saying `Uncaught (in promise)`.
I tried wrapping the selectors with a `try..catch`, but to no avail.

Example:
```js
const selector1 = selector({
	key: "selector-1",
	get: async () => {
		const response = await.someAsyncAPICall();

		return response;
	},
});

const selector2 = selector({
	key: "selector-2",
	get: async ({ get }) => {
		const selector1Response = get(selector1);
		const response = await.someAsyncAPICallWithParams(selector1Response);

		return response;
	},
});


const selector3 = selector({
	key: "selector-3",
	get: async ({ get }) => {
		const selector2Response = get(selector2);
		const response = await.someAsyncAPICallWithParams(selector2Response);

		return response;
	},
});
```

**Comment by nev-r** [comment]
experiencing the same? similar?? in 0.1.2 right now

```ts
const selectorA = atom({
// [...]
```
```ts
const asyncSelector1 = selector({
  get: async ({ get }) => {
    const value = get(selectorA);
    return value && await asyncOperation1(value);
// [...]
```
```ts
const asyncSelector2 = selector({
  get: async ({ get }) => {
    const asyncValue = get(asyncSelector1);
    return asyncValue && await asyncOperation2(asyncValue);
// [...]
```
```ts
const asyncSelector3 = selector({
  get: async ({ get }) => {
    const asyncValue = get(asyncSelector2);
    return asyncValue && await asyncOperation3(asyncValue);
// [...]
```
i am rendering all 3 of these, wrapped in react.suspense, and they are correctly serially filling themselves in. functionality is all right.

the async calls are correctly each running only once, returning the expected data, and not erroring within their function.

however there's an uncaught rejection that's throwing, without enough stack information for me to troubleshoot.
it can be caught by wrapping `get(asyncSelector2)` in a `try` block.

**Comment by justintoman** [comment]
You don't need the `async` in the 2nd and 3rd selectors. If you remove those it works.

**Comment by kislakiruben** [comment]
@justintoman well, I do need them if I'm doing some other API calls in those selectors as well.

**Comment by drarmstr** [comment]
@csantos42 - Something to evaluate with the new selector for the #103 fix.

**Comment by wayou** [comment]
Any progress?

**Comment by drarmstr** [comment]
> Any progress?

It hasn't been enabled for the default open source build just yet, but should be shortly.  You could test it if you want to tweak nightly yourself.  @csantos42 have you had a chance to confirm this issue with that implementation?

**Comment by drarmstr** [comment]
Resolved with 0.2.0
